Update distributed/CONTRIBUTING.md to remove ProcessGroupAgent references and add test instructions

H-Huang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #78658 Reenable assert after test update
* **#78625 Update distributed/CONTRIBUTING.md to remove ProcessGroupAgent references and add test instructions**

ProcessGroupAgent was deprecated so these references to the tests should be removed from comments and docs.

cc @yaroslavvb

H-Huang committed:
Update distributed/CONTRIBUTING.md to remove ProcessGroupAgent references and add test instructions

[ghstack-poisoned]

mrshenli reviewed (approved):
Thanks for updating

H-Huang committed:
Update on "Update distributed/CONTRIBUTING.md to remove ProcessGroupAgent references and add test instructions"


ProcessGroupAgent was deprecated so these references to the tests should be removed from comments and docs.

cc @yaroslavvb 

[ghstack-poisoned]

albanD commented on torch/csrc/autograd/python_variable.cpp:
```diff
@@ -462,37 +462,6 @@ static int THPVariable_clear(THPVariable* self) {
     //        because Tensor asked us to (it's already destructing).
 
     if (!self->cdata.unsafeIsBorrowed()) {
-      // TODO: empirically, on OS X this assert appears to be untrue
```
Why did you remove this comment?

H-Huang replied:
I removed this for this PR and reenabled the assert in a separate PR (#78658)

H-Huang committed:
Update on "Update distributed/CONTRIBUTING.md to remove ProcessGroupAgent references and add test instructions"


ProcessGroupAgent was deprecated so these references to the tests should be removed from comments and docs.

cc @yaroslavvb 

[ghstack-poisoned]

albanD reviewed (approved):
Updated version sounds good!
